### ISSUE TITLE
Chapter 2 - webserver cluster - Terraform warning

### ISSUE THREAD

**Issue by martin-flower** [brikis98/terraform-up-and-running-code#43]
`terraform plan` displays the following in `02-intro-to-terraform-syntax/webserver-cluster`

```
Warning: "condition.0.values": [DEPRECATED] use 'host_header' or 'path_pattern' attribute instead
  on main.tf line 113, in resource "aws_lb_listener_rule" "asg":
 113: resource "aws_lb_listener_rule" "asg" {

```

```
Terraform v0.12.10
+ provider.aws v2.43.0
macOS 10.15.2
```

How do I get rid of this message?

**Comment by brikis98** [comment]
This was a change introduced in https://github.com/terraform-providers/terraform-provider-aws/pull/8268, which was released in [AWS Provider 2.42.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2420-december-13-2019), which came out after the second edition of the book was published. There are two ways you can get rid of this warning:

1. Do as the warning suggests and switch from this format:

    ```hcl
    condition {
      field  = "path-pattern"
      values = ["*"]
    }
    ```

    To this format:

    ```hcl
    condition {
      path_pattern {
        values = ["*"]
      }
    }
    ```

1. Pin yourself to the older AWS provider version (this is not a good long-term fix):

    ```hcl
    provider "aws" {
      version = "= 2.41.0"
    }
    ```

**Comment by martin-flower** [comment]
Thanks Yevgeniy! Very helpful to not only give the solution, but also the explanation. In order to have repeatable build we probably need to pin the terraform and aws-provider versions. Just need to make sure we upgrade regularly.
